### PR TITLE
Workload controller manager needs to die sooner #174

### PR DESCRIPTION
**Pull request by Hong-Chang**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Currently, controller instance record in registry will be automatically deleted after 5 minutes when it could not renew its lease. This is designed for a very stable environment. However, initial testing often has stability problem and caused workload controller manager restart. Further investigation also found api server does lease renewal every 10 seconds. This change will allow problematic controller instances to be discovered much faster.

**Which issue(s) this PR fixes**:
Workload controller manager needs to die sooner #174

**Special notes for your reviewer**:
Please suggest configured values.
I introduced type ControllerConfig, so renamed original ControllerConfig to ControllerConfigMap.

**Does this PR introduce a user-facing change?**:
No

### CODE REVIEW

File: cmd/workload-controller-manager/config/controllerconfig.json [hunk 1]
```diff
@@ -2,15 +2,18 @@
     "controllers": [
         {
             "type":    "node",
-            "workers": 5
+            "workers": 5,
+            "reportHealthIntervalInSecond":  20
```
Comment (Sindica) [inline cmd/workload-controller-manager/config/controllerconfig.json:6]: Heath report interval shall be consistent. Please keep a single value and don't duplicate necessary

File: pkg/cloudfabric-controller/deployment/deployment_controller.go [hunk 2]
```diff
@@ -155,7 +156,7 @@ func NewDeploymentController(dInformer appsinformers.DeploymentInformer, rsInfor
 }
 
 // Run begins watching and syncing.
-func (dc *DeploymentController) Run(workers int, stopCh <-chan struct{}) {
+func (dc *DeploymentController) Run(controllerConfig config.ControllerConfig, stopCh <-chan struct{}) {
```
Comment (Sindica) [inline pkg/cloudfabric-controller/deployment/deployment_controller.go:159]: Run() function has single signature as the original. Try to keep it the same. We are trying to make the migration of controllers from Kube CM to workload CM as easy as possible

**Comment by Sindica** [comment]
+ joe for configuration change review

File: pkg/registry/core/controllerinstance/storage/storage.go [hunk 3]
```diff
@@ -33,7 +33,7 @@ type REST struct {
 	*genericregistry.Store
 }
 
-const TTL = 300 // time-to-live in seconds
+const TTL = 60 // time-to-live in seconds
```
Comment (Sindica) [inline pkg/registry/core/controllerinstance/storage/storage.go:36]: Can we change this to 30 seconds?

File: cmd/workload-controller-manager/config/controllerconfig.json (hunk 1 above)
Comment (Sindica) [reply cmd/workload-controller-manager/config/controllerconfig.json:6]: Perhaps do 10 second per report as API server renew lease every 10 seconds.

**Comment by Hong-Chang** [comment]
Created another pr to replace this one.

**Pull request closed without merging**